Sources/NIO/Embedded.swift

```diff
@@ -618,14 +618,23 @@ public final class EmbeddedChannel: Channel {
     /// - parameters:
     ///     - handler: The `ChannelHandler` to add to the `ChannelPipeline` before register or `nil` if none should be added.
     ///     - loop: The `EmbeddedEventLoop` to use.
-    public init(handler: ChannelHandler? = nil, loop: EmbeddedEventLoop = EmbeddedEventLoop()) {
+    public convenience init(handler: ChannelHandler? = nil, loop: EmbeddedEventLoop = EmbeddedEventLoop()) {
+        let handlers = handler == nil ? [] : [handler!]
```

Comment:
Might be nicer to avoid the `!`:

```suggestion
        let handlers = handler.map { [$0] } ?? []
```